scripts/jenkins/Jenkinsfile

```diff
@@ -109,241 +62,292 @@ CMAKE_FLAGS_BUILD_ALL = [
-            )
-            def frontend = docker.image(
-                namingStrategyHomepage(DOCKER_IMAGES.homepage_frontend)
-            )
-            backend.pull()
-            frontend.pull()
-
-            sh "docker stop -t 5 ${backend_name} || /bin/true"
-            sh "docker rm ${backend_name} || /bin/true"
-            backend.run("""\
-                -e VIRTUAL_HOST=restapi.libelektra.org \
-                -e LETSENCRYPT_HOST=restapi.libelektra.org \
-                -e LETSENCRYPT_EMAIL=jenkins@hub.libelektra.org \
-                --name ${backend_name} \
-                --network=frontend_default \
-                --restart=always"""
-            )
-
-            sh "docker stop -t 5 ${frontend_name} || /bin/true"
-            sh "docker rm ${frontend_name} || /bin/true"
-            frontend.run("""\
-                -e VIRTUAL_HOST=www.libelektra.org \
-                -e LETSENCRYPT_HOST=www.libelektra.org \
-                -e LETSENCRYPT_EMAIL=jenkins@hub.libelektra.org \
-                --name ${frontend_name} \
-                --network=frontend_default \
-                --restart=always"""
-            )
-        }
-    }
+maybeStage("Deploy Homepage", isMaster()) {
+    deployHomepage()
 }
 
 /*****************************************************************************
  * Stage Generators
  *****************************************************************************/
-def generate_docker_build_stages() {
-    def tasks = [:]
 
+/* Populate DOCKER_IMAGES with data
+ *
+ * For this we need a checkout of the scm to generate the hash for the
+ * Dockerfiles which indicates if a rebuild of the images is needed
+ */
+def dockerInit() {
+    node("master") {
+        echo "Processing DOCKER_IMAGES"
+        checkout scm
+        DOCKER_IMAGES.sid = createDockerImageDesc(
+            "debian-sid", this.&idTesting,
+            "./scripts/docker/debian/sid",
+            "./scripts/docker/debian/sid/Dockerfile"
+        )
+        DOCKER_IMAGES.stretch = createDockerImageDesc(
+            "debian-stretch", this.&idTesting,
+            "./scripts/docker/debian/stretch",
+            "./scripts/docker/debian/stretch/Dockerfile"
+        )
+        DOCKER_IMAGES.stretch_minimal = createDockerImageDesc(
+            "debian-stretch-minimal", this.&idTesting,
+            "./scripts/docker/debian/stretch",
+            "./scripts/docker/debian/stretch/Dockerfile.minimal"
+        )
+        DOCKER_IMAGES.stretch_doc = createDockerImageDesc(
+            "debian-stretch-doc", this.&idTesting,
+            "./scripts/docker/debian/stretch",
+            "./scripts/docker/debian/stretch/Dockerfile.doc"
+        )
+        DOCKER_IMAGES.jessie = createDockerImageDesc(
+            "debian-jessie", this.&idTesting,
+            "./scripts/docker/debian/jessie",
+            "./scripts/docker/debian/jessie/Dockerfile"
+        )
+        DOCKER_IMAGES.xenial = createDockerImageDesc(
```

Comment:
:+1: please also mention this in the release notes. Finally we build something else next to Debian.

Reply:
we do so since quite some time already...

Reply:
With *we* I was referring to our infrastructure. What is our Jenkins building since quite some time already?

It has not been in the release notes.

Reply:
It has been in the last big PR. Did not find it noteworthy as we did debian buidls (test-docker) before even though that had other problems

Reply:
Yes, seems like you find many nice achievements (by yourself) not noteworthy :wink:

Currently our goal is to have release notes that mention all our changes with 3 or 5 of the changes highlighted (to be eye catchers to get interest). And then there are specific sections with stuff relevant for maintainers and Elektra's developers.